nosnoc/problem.py

```diff
@@ -604,21 +613,30 @@ def create_complementarity_constraints(self, sigma_p, tau, Uk, s_elastic):
         opts = self.opts
         if opts.dcs_mode == DcsMode.PDS:
             for j in range(opts.n_s):
-                lam_j = self.Lambda(stage=j)
+                lam_j = self.Lambda(stage = j)
+                lam_prev_j = self.Lambda(stage=j-1)
```

Comment:
This should get the previous Finite Element not the previous stage.

Reply:
Take a look at this code for an example of what yours should be doing. 
You of course need to replace `Theta` with the evaluation of `c_pds` at `X_fe`. 
You would also need to add the previous FE's `c_pds` complement with `Lambda(stage=k)`
```python
for j in range(opts.n_s):
    # cross comp with previous FE
    self.create_complementarity([self.Theta(stage=j)], self.prev_fe.Lambda(stage=-1), sigma_p, tau, s_elastic)
    for jj in range(opts.n_s):
        # within fe
        self.create_complementarity([self.Theta(stage=j)], self.Lambda(stage=jj), sigma_p, tau, s_elastic)
```